internal/http/services/owncloud/ocgraph/drive_permissions.go

```diff
@@ -318,6 +344,69 @@ func (s *svc) updateSharePermissions(ctx context.Context, w http.ResponseWriter,
 	_ = json.NewEncoder(w).Encode(lgPerm)
 }
 
+func (s *svc) updateOCMSharePermissions(ctx context.Context, w http.ResponseWriter, ocmshare *ocm.Share, lgPerm *libregraph.Permission, resourceId *providerpb.ResourceId) {
```

Comment:
I wonder if at this point we can pull out some of this code since we have now at least a part of this three times ... but not sure how easy that would be